sql/limit.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Tamir Duberstein (tamird@gmail.com)
+
+package sql
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/cockroachdb/cockroach/sql/parser"
+)
+
+// limit constructs a limitNode based on the LIMIT and OFFSET clauses.
+func (*planner) limit(n *parser.Select, p planNode) (planNode, error) {
+	if n.Limit == nil {
+		return p, nil
+	}
+
+	var limit, offset int
+
+	data := []struct {
+		name       string
+		src        parser.Expr
+		dst        *int
+		defaultVal int
+	}{
+		{"LIMIT", n.Limit.Count, &limit, math.MaxInt64},
```

Comment:
Mixing `math.MaxInt64` and `int` makes me uneasy. Perhaps use `int64` for the types of `limit` and `offset`.